Detectors/DCS/src/DCSProcessor.cxx

```diff
@@ -0,0 +1,553 @@
+      foundBools = 0, foundStrings = 0, foundTimes = 0, foundBinaries = 0;
+
+  // char type
+  foundChars = processArrayType(mAliaseschars, DeliveryType::RAW_CHAR, map, mLatestTimestampchars, mDpscharsmap);
+
+  // int type
+  foundInts = processArrayType(mAliasesints, DeliveryType::RAW_INT, map, mLatestTimestampints, mDpsintsmap);
+
+  // double type
+  foundDoubles = processArrayType(mAliasesdoubles, DeliveryType::RAW_DOUBLE, map, mLatestTimestampdoubles, mDpsdoublesmap);
+
+  // UInt type
+  foundUInts = processArrayType(mAliasesUints, DeliveryType::RAW_UINT, map, mLatestTimestampUints, mDpsUintsmap);
+
+  // Bool type
+  foundBools = processArrayType(mAliasesbools, DeliveryType::RAW_BOOL, map, mLatestTimestampbools, mDpsboolsmap);
+
+  // String type
+  foundStrings = processArrayType(mAliasesstrings, DeliveryType::RAW_STRING, map, mLatestTimestampstrings, mDpsstringsmap);
+
+  // Time type
+  foundTimes = processArrayType(mAliasestimes, DeliveryType::RAW_TIME, map, mLatestTimestamptimes, mDpstimesmap);
+
+  // Binary type
+  foundBinaries = processArrayType(mAliasesbinaries, DeliveryType::RAW_BINARY, map, mLatestTimestampbinaries, mDpsbinariesmap);
+
+  if (!isDelta) {
+    if (foundChars != mAliaseschars.size())
+      LOG(INFO) << "Not all expected char-typed DPs found!";
+    if (foundInts != mAliasesints.size())
+      LOG(INFO) << "Not all expected int-typed DPs found!";
+    if (foundDoubles != mAliasesdoubles.size())
+      LOG(INFO) << "Not all expected double-typed DPs found!";
+    if (foundUInts != mAliasesUints.size())
+      LOG(INFO) << "Not all expected uint-typed DPs found!";
+    if (foundBools != mAliasesbools.size())
+      LOG(INFO) << "Not all expected bool-typed DPs found!";
+    if (foundStrings != mAliasesstrings.size())
+      LOG(INFO) << "Not all expected string-typed DPs found!";
+    if (foundTimes != mAliasestimes.size())
+      LOG(INFO) << "Not all expected time-typed DPs found!";
+    if (foundBinaries != mAliasesbinaries.size())
+      LOG(INFO) << "Not all expected binary-typed DPs found!";
+  }
+
+  // filling CCDB info to be sent in output
+  std::map<std::string, std::string> md;
+  prepareCCDBobject(mccdbSimpleMovingAverage, mccdbSimpleMovingAverageInfo, mName + "/TestDCS/SimpleMovingAverageDPs", mTF, md);
+
+  LOG(DEBUG) << "Size of unordered_map for CCDB = " << mccdbSimpleMovingAverage.size();
+  LOG(DEBUG) << "CCDB entry for TF " << mTF << " will be:";
+  for (const auto& i : mccdbSimpleMovingAverage) {
+    LOG(DEBUG) << i.first << " --> " << i.second;
+  }
+
+  return 0;
+}
+
+//__________________________________________________________________
+
+int DCSProcessor::processDP(const std::pair<DPID, DPVAL>& dpcom)
+{
+
+  // processing single DP
+
+  DPID dpid = dpcom.first;
+  DeliveryType type = dpid.get_type();
+
+  // first we check if the DP is in the list for the detector
+  if (type == DeliveryType::RAW_CHAR) {
+    auto it = std::find(mAliaseschars.begin(), mAliaseschars.end(), dpid);
```

Comment:
this search is actually redundant (as we discussed), it will be much faster to use ``unordered_map<alias, index>`` to check if the processed needs this alias.

Reply:
The idea here is to see if this alias was actually really defined for the DCSProcessor. If I create the unordered_map that you propose, it is true that I can check if an alias should be processed or skipped with the "[]", but if I don't use find but "[]", then in case the alias is not there (--> there is a misconfiguration), it will be created. This is not what I want. I want to check that the configuration of the DCSProcessor matches with the list of aliases that in the DataProcessor it is expected to process. 

I can add the unordered_map that you propose, but with a different meaning --> checking if the DP (which somewhere else was confirmed to be in the detector configuration) needs to be indeed processed, or should be skipped (even though I would not know why, to decide to skip it, one would need to read it, so process it). 

Do yo agree?

Reply:
I mean the unordered_map exactly for that purpose: to check (using find, which is mush faster for the map than for the vector) if the DP is requested by this processor.
Additionally, the value associated with the key can be used as a slot to fill (up to detector)